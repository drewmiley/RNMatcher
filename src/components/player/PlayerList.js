import React, { Component } from 'react';
import {
    Text,
    View
} from 'react-native';

import PlayerView from './PlayerView';

export default class PlayerList extends Component {
    constructor(props) {
        super(props);
        console.log(props);
        this.shouldComponentUpdate = (nextProps, nextState) => {
            return this.props.players !== nextProps.players;
        };
    };
    render() {
        return <View>
            {this.props.players.map(player =>
                <PlayerView
                    key={player.get('name')}
                    name={player.get('name')}
                    rating={player.get('rating')}
                    {...this.props} />
            )}
        </View>
    };
};