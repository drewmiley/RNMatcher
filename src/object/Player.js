import {Map} from 'immutable';

export default Player = (id, name, rating, map) => {
    const player = {id, name, rating};
    return map ? Map(player) : player;
}