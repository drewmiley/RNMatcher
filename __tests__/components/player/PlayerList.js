import 'react-native';
import { List } from 'immutable';
import React from 'react';
import renderer from 'react-test-renderer';

import PlayerList from '../../../src/components/player/PlayerList';

import Player from '../../../src/object/Player';

describe('PlayerList', () => {

    describe('Rendering', () => {

        it('should match the snapshot', () => {
            const players = List.of(
                new Player(1, 'Drew', 10, true),
                new Player(2, 'James', 20, true),
                new Player(3, 'Miley', 30, true)
            );
            const component = renderer.create(
                <PlayerList players={players} />
            );            
            expect(component).toMatchSnapshot();
        });

    });

});