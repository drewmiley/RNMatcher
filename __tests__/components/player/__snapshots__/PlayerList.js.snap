exports[`PlayerList Rendering should match the snapshot 1`] = `
<View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Drew
       - 
      10
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
        Match
      </Text>
    </View>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      James
       - 
      20
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
        Match
      </Text>
    </View>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Miley
       - 
      30
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
        Match
      </Text>
    </View>
  </View>
</View>
`;
